src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -154,20 +166,37 @@ private ConstantValue DefaultSyntaxValue
             }
         }
 
+        protected ConstantValue MakeDefaultExpression(DiagnosticBag diagnostics, Binder binder)
+        {
+            var result = MakeDefaultExpression(diagnostics, binder, out var convertedExpression);
+
+            if (binder?.IsSemanticModelBinder != true && (this.ContainingSymbol as MethodSymbol)?.MethodKind != MethodKind.LocalFunction)
+            {
+                // It is okay to call this multiple times (in a multithreaded environment),
+                // as the only thing VariableUsePass is add/remove members from sets,
+                // which can be performed multiple times with no further effect.
+                new FieldUsePass(this.DeclaringCompilation.SourceAssembly).Visit(convertedExpression);
```

Comment:
>new FieldUsePass(this.DeclaringCompilation.SourceAssembly).Visit(convertedExpression); [](start = 16, length = 86)

It doesn't look like this call site is handling conversion of stack overflow into diagnostics.